mat64/triangular.go

```diff
@@ -0,0 +1,55 @@
+package mat64
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+var (
+	triangular *Triangular
+
+	_ Matrix  = triangular
```

Comment:
I'm following the pattern that exists there now. We can change it if there's a better way, but consistency is better.